src/best_of/generators/markdown_gallery.py

```diff
@@ -0,0 +1,394 @@
+            # table_html += project.name
+            project_md = generate_project_html(project, config, labels)
+            table_html += f'<td valign="top" width="33.3%">{project_md}</td>'
+        table_html += "</tr>"
+    table_html += "</table>"
+    print()
+    return table_html
+
+
+def generate_category_gallery_md(
+    category: Dict, config: Dict, labels: list, title_md_prefix: str = "##"
+) -> str:
+    """Generates markdown gallery for a category, containing tables with projects."""
+    category_md = ""
+
+    if (
+        (
+            config.hide_empty_categories
+            or category.category == default_config.DEFAULT_OTHERS_CATEGORY_ID
+        )
+        and not category.projects
+        and not category.hidden_projects
+    ):
+        # Do not show category
+        return category_md
+
+    # Set up category header.
+    category_md += title_md_prefix + " " + category.title + "\n\n"
+    # TODO: Original line doesn't work if there's no TOC. Replaced it with link
+    #   to title for now but fix this in original repo.
+    # category_md += f'<a href="#contents"><img align="right" width="15" height="15" src="{best_of.default_config.UP_ARROW_IMAGE}" alt="Back to top"></a>\n\n'
+    category_md += f'<a href="#----best-of-streamlit----"><img align="right" width="15" height="15" src="{default_config.UP_ARROW_IMAGE}" alt="Back to top"></a>\n\n'
+    if category.subtitle:
+        category_md += "_" + category.subtitle.strip() + "_\n\n"
+
+    if category.projects:
+        # Show top projects directly (in a html table).
+        num_shown = config.get("projects_per_category", 6)
+        table_html = generate_table_html(category.projects[:num_shown], config, labels)
+        category_md += table_html + "\n\n"
+
+        # Hide other projects in an expander.
+        if len(category.projects) > num_shown:
+            hidden_table_html = generate_table_html(
+                category.projects[num_shown:], config, labels
+            )
+            category_md += f'<br><details align="center"><summary><b>Show {len(category.projects) - num_shown} more for "{category.title}"</b></summary><br>{hidden_table_html}</details>\n\n'
+
+    # This is actually not used here (because all projects are set to show:
+    # True) but it's left here from the original `best_of.generate_category_md` function
+    # for completeness.
+    if category.hidden_projects:
+        category_md += (
+            "<details><summary>Show "
+            + str(len(category.hidden_projects))
+            + " hidden projects...</summary>\n\n"
+        )
+        for project in category.hidden_projects:
+            project_md = markdown_list.generate_project_md(
+                project, config, labels, generate_body=False
+            )
+            category_md += project_md + "\n"
+        category_md += "</details>\n"
+    # print(category_md)
+    return "<br>\n\n" + category_md
+
+
+def process_md_link(text: str) -> str:
```

Comment:
Is it possible to reuse the `process_md_link` of the `markdown_list`? We might also move this to utils.

Reply:
Yup, changed it.